src/rpc/mod.rs

```diff
@@ -289,6 +290,61 @@ pub fn start(addr: &SocketAddr, client_addr: &SocketAddr, base_path: Option<Path
         });
     }
 
+    let address_dir = chain
+        .get_path("addressbook".to_string())
+        .expect("Expect directory for address book");
+
+    let addressbook = Arc::new(Addressbook::new(address_dir));
+
+    {
+        let addressbook = addressbook.clone();
+
+        io.add_async_method("emerald_addressBook",
+                            move |_| futures::finished(Value::Array(addressbook.list())).boxed());
+    }
+
+    {
+        let addressbook = addressbook.clone();
+
+        io.add_async_method("emerald_addAddress", move |p| match p {
```

Comment:
Yeah, it is not optimal. I think the methods will contain more functionality in the future so I will leave them separate for now.